### DISCUSSION TITLE
useMutationState is returning data undefined initially

### DISCUSSION THREAD

**Discussion by Chiratna** [TanStack/query#6723] [Q&A]
Suppose there are 2 screens in a react-native app ( **Screen A**, **Screen B** ) both on the same Stack. I am on **Screen A**,  **Screen B** still has not been pushed to the stack. I call a mutation say `usePostMutation` and on success of that mutation I am navigating to **Screen B**. 

In **Screen B** I am using `useMutationState` to access the data of `usePostMutation`. I am able to get the data but initially it is `undefined` and then I am able to see the actual data. 

My question is since I am navigating on to the **Screen B** `onSuccess` of the `usePostMutation`  why my data which I am getting using `useMuatationState`  is undefined initially ??

**Comment by TkDodo** [comment]
shouldn't be; show a reproduction, preferably in codesandbox, because it's likely not react-native specific

**Reply by chrism1996** [reply]
Getting the same issue. I'm not able to get the data but I can see it in the devtools, it's always as pending and data is undefined. In my case is with sidebars

**Reply by chrism1996** [reply]
Issue fixed using mutateAsync instead of mutate, should we get an example about this specific section that helped me to fix the issue 
![image](https://github.com/user-attachments/assets/af485487-f90a-40ee-bf31-81d846e90e7b)

**Reply by TkDodo** [reply]
There shouldn’t be a difference between `mutate` and `mutateAsync` as far as `useMutationState` is concerned. `mutate` is really just: `(vars) => mutateAsync(vars).then(noop).catch(noop)`

I’m still waiting for a reproduction to look into it.
